src/prefect/infrastructure/provisioners/__init__.py

```diff
@@ -2,13 +2,14 @@
 
 from prefect.client.orchestration import PrefectClient
 from prefect.infrastructure.provisioners.modal import ModalPushProvisioner
-from .cloud_run import CloudRunPushProvisioner
+from .cloud_run import CloudRunPushProvisioner, CloudRunV2PushProvisioner
 from .container_instance import ContainerInstancePushProvisioner
 from .ecs import ElasticContainerServicePushProvisioner
 import rich.console
 
 _provisioners = {
     "cloud-run:push": CloudRunPushProvisioner,
+    "cloud-run-v2:push": CloudRunV2PushProvisioner,
```

Comment:
Since the `CloudRunPushProvisioner` and `CloudRunV2PushProvisioner` are identical other than their name, I think we can completely reuse the `CloudRunPushProvisioner` for the `cloud-run-v2:push` work pool. In that case, I think we'll only need a test to confirm that `get_infrastructure_provisioner_for_work_pool_type` accepts `cloud-run-v2:push` as a valid type.

```suggestion
    "cloud-run-v2:push": CloudRunPushProvisioner,
```

Reply:
Good though on just using single provisioner. Pushed those changes.
The tests for v2 load the v2 job template and then use that. Might be good to keep that test coverage. What do you think?

Reply:
Yeah, I think it's worth having test coverage that makes sure the existing provisioner works with the default V2 base job template.